crates/types/src/entities/coins/coin.rs

```diff
@@ -40,21 +40,36 @@ pub struct Coin {
 impl Coin {
     /// Compress the coin to minimize the serialized size.
     pub fn compress(self) -> CompressedCoin {
-        CompressedCoin {
+        CompressedCoin::V1(CompressedCoinV1 {
             owner: self.owner,
             amount: self.amount,
             asset_id: self.asset_id,
             maturity: self.maturity,
             tx_pointer: self.tx_pointer,
-        }
+        })
     }
 }
 
 /// The compressed version of the `Coin` with minimum fields required for
 /// the proper work of the blockchain.
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum CompressedCoin {
+    /// CompressedCoin Version 1
+    V1(CompressedCoinV1),
+}
+
+impl Default for CompressedCoin {
```

Comment:
It is better to only allow to use `Default` implementation in the tests. In the real code we want to create explicitly specific version